src/main.rs

```diff
@@ -276,6 +278,69 @@ fn main() {
         }
     };
 
+    if args.edit_page && args.edit_patch {
+        print_error(
+            enable_styles,
+            &anyhow!("Don't edit-patch and edit-page at the same time"),
+        );
+        process::exit(1);
+    }
+
+    if args.edit_patch || args.edit_page {
+        if args.command.is_empty() {
+            if args.edit_patch {
+                print_error(
+                    enable_styles,
+                    &anyhow!(
+                        "
+No command given
+
+Sample usage :
+    tldr --edit-patch <cmd>",
+                    ),
+                );
+            }
+            if args.edit_page {
+                print_error(
+                    enable_styles,
+                    &anyhow!(
+                        "
+No command given
+
+Sample usage :
+    tldr --edit-page <cmd>",
+                    ),
+                );
+            }
+            process::exit(1);
+        }
+        let command = args.command.join("-").to_lowercase();
+        let custom_page_dir = config
+            .directories
+            .custom_pages_dir
+            .as_ref()
+            .map(PathWithSource::path);
+        if let Some(custom_page_dir) = custom_page_dir {
+            let _ = create_dir_all(custom_page_dir);
+
+            let file_name = if args.edit_patch {
+                format!("{command}.patch.md")
+            } else {
+                format!("{command}.page.md")
+            };
+            let custom_page_path = custom_page_dir.join(file_name);
+            println!("Editing {custom_page_path:?}");
+            let editor = env::var("EDITOR").context("env `EDITOR` not set").unwrap();
+            let _ = std::process::Command::new(editor)
```

Comment:
Right now, this does not work if people set EDITOR to something that includes arguments, for example `EDITOR="nvim --clean" tldr --edit-page foo`. I checked a couple of programs for how they handle this. Both `git config --edit` and `sudoedit` just spawn nvim without arguments, omitting the `--clean`. I suppose that having arguments in EDITOR is not a common thing then.

I think that crashing is better than omitting the arguments, so the current handling of EDITOR is fine. However, we should handle and print errors returned here.

Additionally, we should think about adding some documentation for this behavior, but let's postpone this until the code is ready.

Reply:
we can use shlex to parse

Reply:
Nah, as I said, the current behavior is probably exactly what we want to do. As long as the behavior is documented, I think we are good.

You could add a section at the end of `docs/src/usage_custom_pages.md` to highlight the feature and we can phrase it so that the behavior is clear